aiohttp_asgi_connector/transport.py

```diff
@@ -127,6 +127,14 @@ async def send(message: "MutableMapping[str, Any]") -> None:
     def _encode_response(
         self, status: int, headers: "List[Tuple[bytes, bytes]]", body: bytearray
     ) -> bytes:
+        # add content-length if not present, which happens with streamed responses
+        found = False
+        for (name, value) in headers:
+            if name.lower() == b"content-length":
+                found = True
+        if not found:
+            headers.append((b"content-length", str(len(body)).encode()))
+
```

Comment:
part of me wonders whether i took the right approach here originally, by which i mean ignoring chunked & compressed response bodies

supporting those would make this more complicated, but i can see the value in doing it to avoid serialization problems like this... also if users actually _do_ want to test their intended server output...

Reply:
https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope

> Servers are responsible for handling inbound and outbound chunked transfer encodings. A request with a chunked encoded body should be automatically de-chunked by the server and presented to the application as plain body bytes; a response that is given to the server with no Content-Length may be chunked as the server sees fit.

https://asgi.readthedocs.io/en/latest/specs/www.html#response-start-send-event

> You may send a Transfer-Encoding header in this message, but the server must ignore it. Servers handle Transfer-Encoding themselves, and may opt to use Transfer-Encoding: chunked if the application presents a response that has no Content-Length set.

The way I read it is, the proper way is to implement chunked transfer encoding, but the approach implemented here is also valid, although it wouldn't support endpoints that use long polling.